File: ui/src/app/models/dietary-profile.model.ts
export interface DietaryProfile {
  dietaryProfileResourceId: string;
  subjectId: string;
  profileName: string;
  isDefault: boolean;
  restrictions: DietaryRestriction[];
  avoidedIngredients: AvoidedIngredient[];
  createdDate: Date;
  lastModifiedDate: Date;
}

export interface DietaryRestriction {
  restrictionType: string; // 'Allergy', 'Intolerance', 'DietaryChoice'
  restrictionCode: string;
  severity: string; // 'Strict', 'Flexible'
}

export interface AvoidedIngredient {
  avoidedIngredientId: number;
  ingredientName: string;
  reason?: string;
}

export interface UpdateDietaryProfile {
  profileName: string;
  isDefault: boolean;
}

export interface AddDietaryRestriction {
  restrictionCode: string;
  restrictionType: string;
  severity: string;
}

export interface AddAvoidedIngredient {
  ingredientName: string;
  reason?: string;
}

export interface DietaryConflictCheck {
  canEat: boolean;
  conflicts: DietaryConflict[];
}

export interface DietaryConflict {
  ingredientIndex: number;
  ingredientText: string;
  restrictionCode: string;
  restrictionType: string;
  severity: string;
  message: string;
}

File: src/RecipeVault.WebApi/Models/Responses/DietaryProfileModel.cs
#pragma warning disable CS1591 // Missing XML comments

using System;
using System.Collections.Generic;

namespace RecipeVault.WebApi.Models.Responses {
    public class DietaryProfileModel {
        public Guid DietaryProfileResourceId { get; set; }
        public Guid SubjectId { get; set; }
        public string ProfileName { get; set; }
        public bool IsDefault { get; set; }
        public List<DietaryRestrictionModel> Restrictions { get; set; } = new();
        public List<AvoidedIngredientModel> AvoidedIngredients { get; set; } = new();
        public DateTime CreatedDate { get; set; }
        public DateTime LastModifiedDate { get; set; }
    }

    public class DietaryRestrictionModel {
        public string RestrictionType { get; set; }
        public string RestrictionCode { get; set; }
        public string Severity { get; set; }
    }

    public class AvoidedIngredientModel {
        public int AvoidedIngredientId { get; set; }
        public string IngredientName { get; set; }
        public string Reason { get; set; }
    }

    public class DietaryConflictCheckModel {
        public bool CanEat { get; set; }
        public List<DietaryConflictModel> Conflicts { get; set; } = new();
    }

    public class DietaryConflictModel {
        public int IngredientIndex { get; set; }
        public string IngredientText { get; set; }
        public string RestrictionCode { get; set; }
        public string RestrictionType { get; set; }
        public string Severity { get; set; }
        public string Message { get; set; }
    }
}
